catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -524,12 +539,70 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+
+          final Metacard workspace = getWorkspaceMetacard(id);
+          Attribute queries = workspace.getAttribute(WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+
+          if (queries != null) {
+            String[] queryIds = queries
+                .getValues()
+                .stream()
+                .map(String.class::cast)
+                .toArray(String[]::new);
+
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
+          }
+
           catalogFramework.delete(new DeleteRequestImpl(id));
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    post(
+        "/query",
+        APPLICATION_JSON,
+        (req, res) -> {
+          Map<String, Object> body = JsonFactory.create().parser().parseMap(util.safeGetBody(req));
+          Metacard metacard = queryTransformer.transformJsonIntoMetacard(body);
+          Metacard saved = saveMetacard(metacard);
+          return util.metacardToJson(saved.getId());
+        });
+
+    put(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String id = req.params("id");
+          Map<String, Object> body = JsonFactory.create().parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard = queryTransformer.transformJsonIntoMetacard(body);
+          Metacard updated = updateMetacard(id, metacard);
+
+          return util.getMetacardMap(updated);
+        },
+        util::getJson);
+
+    get(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String id = req.params("id");
+          Metacard metacard = util.getMetacard(id);
+          return util.metacardToJson(metacard.getId());
+        });
+
+    delete(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String id = req.params(":id");
+          catalogFramework.delete(new DeleteRequestImpl(id));
+          // TODO Remove workspace metacard association
```

Comment:
I ended up changing this endpoint quite a bit. It now looks like `/query/:workspaceId/:queryId`. Since there's not context for queries to exist outside of their workspace metacard right now you have to pass in a workspace ID.